split.js

```diff
@@ -1,33 +1,42 @@
 var utils = require('./utils')
+var BN = require('bn.js')
+var ext = require('./bn-extensions')
 
 // split utxos between each output, ignores outputs with .value defined
 module.exports = function split (utxos, outputs, feeRate) {
-  if (!isFinite(utils.uintOrNaN(feeRate))) return {}
+  if (!isFinite(utils.bnOrNaN(feeRate))) return {}
 
   var bytesAccum = utils.transactionBytes(utxos, outputs)
-  var fee = feeRate * bytesAccum
+  var fee = ext.mul(feeRate, bytesAccum)
   if (outputs.length === 0) return { fee: fee }
 
   var inAccum = utils.sumOrNaN(utxos)
   var outAccum = utils.sumForgiving(outputs)
-  var remaining = inAccum - outAccum - fee
+  var remaining = ext.sub(inAccum, outAccum, fee)
   if (!isFinite(remaining) || remaining < 0) return { fee: fee }
 
   var unspecified = outputs.reduce(function (a, x) {
     return a + !isFinite(x.value)
```

Comment:
and this

Reply:
looks like they really just wanted to do `outputs.filter(x => !isFinite(x.value)).length`
but wanted to maintain compatibility with old js

Reply:
but it also uses `.every`..so `filter` should have been fine...